amaranth/lib/wiring.py

```diff
@@ -842,7 +834,29 @@ def signature(self):
                         raise SignatureError(f"Member '{name}' is redefined in {base.__module__}.{base.__qualname__}")
                     members[name] = annot
         if not members:
-            raise NotImplementedError(
-                f"Component '{cls.__module__}.{cls.__qualname__}' does not have signature member "
-                f"annotations")
-        return Signature(members)
+            if signature is None:
```

Comment:
The warning above will fire even in `if not members:` case. I think it can be just removed; there are legitimate reasons to have an annotation for some value-castable attribute that is not a part of the signature (maybe it's exposed for debugging and then typed).

Reply:
OK, warning removed.